frontend/components/product-list/sections/FilterableProductsSection/useProductSearchHitPricing.ts

```diff
@@ -0,0 +1,44 @@
+import * as React from 'react';
+import { useAuthenticatedUser } from '@ifixit/auth-sdk';
+import { ProductSearchHit } from '@models/product-list';
+import { computeDiscountPercentage } from '@helpers/commerce-helpers';
+
+export type ProductSearchHitPricing = {
+   price: number;
+   compareAtPrice: number | undefined;
+   isDiscounted: boolean;
+   percentage: number;
+};
+export function useProductSearchHitPricing(
+   product: ProductSearchHit
+): ProductSearchHitPricing {
+   const user = useAuthenticatedUser();
+
+   const proTierPrice = React.useMemo(() => {
```

Comment:
I'd vote for not using `useMemo()` here. I'd be willing to bet that `useMemo()` and creating this closure takes *wayy* longer than just running these if statements every time. They are just property accesses and one `parseFloat()`.